# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1465677425 14400
#      Sat Jun 11 16:37:05 2016 -0400
# Node ID 297d4fc3c5631d50ba8a15d616b0014f1ed9ef50
# Parent  0469ae203a1dfe5c11f9efc499854fa34ff51c33
* str-vec.h, str-vec.cc (string_vector::std_list): New method.

diff --git a/liboctave/util/str-vec.cc b/liboctave/util/str-vec.cc
--- a/liboctave/util/str-vec.cc
+++ b/liboctave/util/str-vec.cc
@@ -193,16 +193,29 @@ string_vector::c_str_vec (void) const
   retval[len] = 0;
 
   for (octave_idx_type i = 0; i < len; i++)
     retval[i] = strsave (elem (i).c_str ());
 
   return retval;
 }
 
+std::list<std::string>
+string_vector::std_list (void) const
+{
+  octave_idx_type len = numel ();
+
+  std::list<std::string> retval;
+
+  for (octave_idx_type i = 0; i < len; i++)
+    retval.push_back (elem (i));
+
+  return retval;
+}
+
 void
 string_vector::delete_c_str_vec (const char * const *v)
 {
   const char * const *p = v;
 
   while (*p)
     delete [] *p++;
 
diff --git a/liboctave/util/str-vec.h b/liboctave/util/str-vec.h
--- a/liboctave/util/str-vec.h
+++ b/liboctave/util/str-vec.h
@@ -108,16 +108,18 @@ public:
   string_vector& append (const std::string& s);
 
   string_vector& append (const string_vector& sv);
 
   std::string join (const std::string& sep = "") const;
 
   char **c_str_vec (void) const;
 
+  std::list<std::string> std_list (void) const;
+
   static void delete_c_str_vec (const char * const*);
 
   std::ostream&
   list_in_columns (std::ostream&, int width = 0,
                    const std::string& prefix = "") const;
 };
 
 #endif
